### PR TITLE
Custom "Miner" Payouts Feature

### LINKED ISSUE #206: declare 3Bot foundation fees as explicit fees in the parent block
Currently these fees are 100% implicit, with as a consequence that the fee structure (condition for payout and amount) with as a consequence is _never_ registered on the blockchain.

Computing the fees is simple enough currently, as it can be assumed (as enforced by the consensus engine) that the fee is exactly as much as is required, thus re-computing the required fee should give you the correct amount.

Still, not having it registered on the chain, does mean that every client (full or light) has to know for every blockheight what the payout foundation (pool) address is for fees, while this knowledge would only be required for block creators should these conditions be part of the created blocks. This is still acceptable, in terms of byte size, if we ensure to merge the fees of each transaction in a block together, as to form just a single payout with the foundation (pool) address active at that moment.

### LINKED ISSUE #237: integrate Registration Fee of ERC20 Address Registration Tx into parent block
For now the fee is burned.

### PR DESCRIPTION
**Pull request by GlenDC**
Move our custom fees to the parent block next to the regular miner payouts:
* 3Bot Fees are now registered as miner payouts in the parent block: closes #206;
* ERC20 Address Registration Fees are now registered as miner payouts in the parent block: closes #237;

Also closes Rivine Issue 
https://github.com/threefoldtech/rivine/issues/479, adding that solution directly as a vendor-overwrite. Once this PR is accepted (and merged), I can move/contribute it back to Rivine, and update the vendor metadata to have it all be in sync (once) again. When that is done I'll close that issue as well.

TLDR: We now have _a_ way for transaction types to register custom miner payouts. The web explorer is already updated as well and is aware of all it.

---

Extra:

screen shot 2018-12-19 at 22 28 35
<center>--</center>
screen shot 2018-12-19 at 22 29 29

> Transaction pages show the fees now with the details of the actual payout linked to it, just as was already the case for the (regular) Transaction Fees

screen shot 2018-12-19 at 22 28 50
<center>--</center>
screen shot 2018-12-19 at 22 30 21

> Block pages now show the Miner Payouts with a more appropriate title and each payout has more details attached to it, so it becomes immediately clear what is what.

screen shot 2018-12-19 at 22 29 47

> Coin Outputs that are actually reward/fee payouts, now show as part of the title what kind of payout it is, with this as an example

screen shot 2018-12-19 at 22 29 58
<center>--</center>
screen shot 2018-12-19 at 22 30 37

> When going to a payout address, you'll find more details for each payout, just as is now also the case with the block pages.

### CODE REVIEW

**Comment by GlenDC** [comment]
Ready for review+merge.
